core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorFactory.java

```diff
@@ -14,13 +14,31 @@
 package io.trino.spi.connector;
 
 import com.google.errorprone.annotations.CheckReturnValue;
+import io.trino.spi.Experimental;
 
 import java.util.Map;
+import java.util.Set;
 
 public interface ConnectorFactory
 {
     String getName();
 
     @CheckReturnValue
     Connector create(String catalogName, Map<String, String> config, ConnectorContext context);
+
+    /**
+     * Returns property names that may contain security-sensitive information.
+     * In addition to properties that are explicitly known to the connector as
+     * security-sensitive, it may also return properties that are unknown or unused.
+     * In other words, if the connector cannot classify a property, it should treat it as
+     * security-sensitive by default for safety.
+     * <p>
+     * The engine uses the properties returned by this method to mask the corresponding
+     * values, preventing the leakage of security-sensitive information.
+     */
+    @Experimental(eta = "2025-12-31")
+    default Set<String> getSecuritySensitivePropertyNames(String catalogName, Map<String, String> config, ConnectorContext context)
```

Comment:
Why does the method take the full config map and not just the names?

Reply:
I don't think this PR should go in until we have more clarity on exactly how it will be used by the engine. An SPI in a vacuum and without usages doesn't make sense to include.

The PR linked in the description is still work in progress, so let's narrow down on that first.

Reply:
> Why does the method take the full config map and not just the names?

We use the values to get security-sensitive properties from Airlift. Here’s an example: https://github.com/trinodb/trino/blob/c214ce6413d9d7b0ba9e9ad5ee3c6175bb13c4c7/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java

Reply:
@martint that's because config properties are dependant on each other and without actual values you don't know which ones will be bound during bootstrap